llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1794,7 +1803,14 @@ static bool eliminateConstraints(Function &F, DominatorTree &DT, LoopInfo &LI,
       if (ReproducerModule && DFSInStack.size() > ReproducerCondStack.size())
         ReproducerCondStack.emplace_back(Pred, A, B);
 
-      Info.transferToOtherSystem(Pred, A, B, CB.NumIn, CB.NumOut, DFSInStack);
+      // If samesign is present on the ICmp, simply transfer the signed system
+      // to the unsigned system, and viceversa.
+      if (HasSameSign)
+        Info.addFact(CmpInst::getFlippedSignednessPredicate(Pred), A, B,
```

Comment:
Should avoid duplicating facts with equality predicates.

Reply:
I would have expected samesign to be illegal on ICmps with equality predicates? What sense does `icmp samesign eq` or `icmp samesign ne` make?

Reply:
It is allowed by spec.